### PR TITLE
Introduce shortcircuits API

### PR DESCRIPTION
**Pull request by obrix**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

Move and adapt shortcircuits api from RTE repository. The adaptation mostly consists in renaming the package and adding the main pom.xml and shortcircuits-api module.

### CODE REVIEW

**Comment by obrix** [comment]
@MioRtia I added you as a reviewer to see if you have any insight about how to make the short-circuits api more consistent with the other API in powsybl-core. For now I took the existing API and just made sure it compiled but I think there is more we can do.

The first task I see is to replace the factory design pattern by a provider design pattern to match what is done elsewhere in the core.

**Review by MioRtia — CHANGES_REQUESTED** [review]
I didn't write it on all the files because it was a bit repetitive but copyrights and authors are missing.
Regarding implementation itself, in my opinion, there are only some convenient static `run` methods missing int `ShortCircuitsAnalysis` to be more user-friendly

**Comment by obrix** [comment]
Thanks for the review I'll add the copyrights and take a look at your suggestions . Concerning the authors I don't have access to the original git repository (just a copy of the code) so I don't think I will be able to retrieve the original authors for each files.

File: shortcircuits-api/src/main/java/com/powsybl/shortcircuits/ShortCircuitsAnalysis.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.network.Network;
+
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * API for short-circuit current computations.
+ */
+public final class ShortCircuitsAnalysis {
```
[the remark this answers is not in the document]
Comment (obrix) [reply shortcircuits-api/src/main/java/com/powsybl/shortcircuits/ShortCircuitsAnalysis.java:22]: I simply added a run method with a single network parameter. I don't know what you mean by checking configuration here as the implementation is a hollow shell at the moment. Feel free to make some adjustments if you think some things are relevant to add.

**Comment by MioRtia** [comment]
> @MioRtia Thanks for the review I'll add the copyrights and take a look at your suggestions . Concerning the authors I don't have access to the original git repository (just a copy of the code) so I don't think I will be able to retrieve the original authors for each files.
> 
> @MioRtia Just to be more explicit, most of the files are added "as they came" from the rte code, if there is no authors given in the files, there was also no author given in the original code.

Thank you for the reply! Don't mind the authors then, I will add them before merging as I think I still have the history of the files.

**Review by MioRtia — APPROVED** [review]

**Review by annetill — APPROVED** [review]
This is for me a first version that should be improved. The first big improvement is that results should be given at the level connectable + side, and not more for a voltage level.

**Pull request merged**